pmm-app/src/pmm-settings/components/PlatformLogin/PlatformLogin.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { FC, useState } from 'react';
+import { useTheme } from '@grafana/ui';
+import { getStyles } from './PlatformLogin.styles';
+import { SignUpProps } from './types';
+import { LoggedIn } from './LoggedIn/LoggedIn';
+import { SignIn } from './SignIn';
+import { SignUp } from './SignUp';
+
+export const PlatformLogin: FC<SignUpProps> = ({ userEmail, getSettings }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  const [isSignInMode, setMode] = useState(true);
+  const toggleMode = () => setMode(!isSignInMode);
```

Comment:
this should be `setMode((currentMode) => !currentMode)`. doing this you're not dependent on `isSignInMode` in `setMode`